modules/puppet/manifests/puppetserver/package.pp

```diff
@@ -0,0 +1,71 @@
+# == Class: puppet::puppetserver::package
+#
+# Install packages for a Puppetserver.
+#
+# === Parameters
+#
+# [*hiera_eyaml_gpg_version*]
+#  Specify the version of hiera_eyaml_gpg to install.
+#
+# [*puppetdb_version*]
+#   Specify the version of puppetdb-terminus to install which should match
+#   the puppetdb installation. Passed in by parent class.
+#
+# [*puppet_sentry_dsn*]
+#   This is the DSN to send puppet reports to.
+#
+class puppet::puppetserver::package(
+  $hiera_eyaml_gpg_version = 'latest',
+  $puppetdb_version = 'present',
+  $puppet_sentry_dsn = undef,
+) {
+  include ::puppet
+
+  ensure_packages(['openjdk-7-jre-headless'])
+
+  package { 'puppetserver':
+    ensure  => installed,
+    require => Package['openjdk-7-jre-headless'],
+  }
+
+  package { 'aws-sdk-core':
+    provider => system_gem,
+  }
+
+  package { ['hiera-eyaml-gpg', 'gpgme']:
+    ensure   => absent,
+    provider => system_gem,
+  }
+
+  package { 'rack':
+    ensure   => '1.0.1',
+    provider => system_gem,
+  }
+
+  package { 'puppetdb-terminus':
+    ensure  => $puppetdb_version,
+  }
+
+  exec { 'puppetserver gem install hiera-eyaml-gpg':
```

Comment:
I think you have to set a fully qualified path or search path when running `execs`